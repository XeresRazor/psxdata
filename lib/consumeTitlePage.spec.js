const path = require('path');
const { JSDOM } = require('jsdom');
const consumeTitlePage = require('./consumeTitlePage');

describe('consumeTitlePage', () => {
  describe('produces expected output under JSDOM', () => {
    it('for TEKKEN 3', (done) => {
      expect.assertions(1);

      JSDOM.fromFile(
        path.join(__dirname, '__fixtures__', 'SLPS-01300.html'),
        { url: 'https://test-psxdata' }
      )
      .then(
        (dom) => {
          expect(consumeTitlePage(dom.window.document)).toMatchSnapshot();
          done();
        },
        (error) => {
          done();
        }
      );
    });

    it('for LEMMINGS', (done) => {
      expect.assertions(1);

      JSDOM.fromFile(
        path.join(__dirname, '__fixtures__', 'SCES-00009.html'),
        { url: 'https://test-psxdata' }
      )
      .then(
        (dom) => {
          expect(consumeTitlePage(dom.window.document)).toMatchSnapshot();
          done();
        },
        (error) => {
          done();
        }
      );
    });
  });
});
