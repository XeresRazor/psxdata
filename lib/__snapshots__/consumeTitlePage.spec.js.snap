// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`consumeTitlePage produces expected output under JSDOM for LEMMINGS 1`] = `
Object {
  "commonTitle": "3D Lemmings",
  "coverLink": "https://test-psxdata/images/covers/P/L/SCES-00009.jpg",
  "date": "1995-10-01",
  "developer": "Clockwork Games Limited",
  "genre": "Puzzle",
  "publisher": "Psygnosis",
  "rating": Array [
    "elspa-3",
    "usk-0",
  ],
  "style": undefined,
  "title": "Lemmings 3D",
}
`;

exports[`consumeTitlePage produces expected output under JSDOM for TEKKEN 3 1`] = `
Object {
  "coverLink": "https://test-psxdata/images/covers/J/T/SLPS-01300.jpg",
  "date": "1998-03-26",
  "developer": "Namco",
  "genre": "Fighting",
  "publisher": "Namco",
  "rating": Array [
    "esrb-t",
    "usk-0",
  ],
  "style": undefined,
  "title": "Tekken 3",
}
`;
